### PR TITLE
improved profiles_fluxes plotting method

### PR DESCRIPTION
**Pull request by TimSlendebroek**
Changes so that it makes:

image

### CODE REVIEW

File: omas/omas_plot.py [hunk 1]
```diff
@@ -1075,7 +1089,7 @@ def TGYRO_profiles_fluxes(ods, time_index=0, fig=None, axes=None, state="None",
         if len(time) == 1:
             time_index = time_index[0]
         else:
-            return ods_time_plot(TGYRO_profiles_fluxes, time, ods, time_index, fig=fig, ax={}, **kw)
+            return ods_time_plot(profiles_fluxes, time, ods, time_index, fig=fig, ax={}, **kw)
```
Comment (orso82) [inline omas/omas_plot.py:1092]: `**kw` is not right. You'll need to explicitly pass all of the arguments to profiles_fluxes.

File: omas/omas_plot.py [hunk 2]
```diff
@@ -1112,68 +1121,87 @@ def sum_density_types(specie_index):
...
     # Fluxes
     if "core_transport" in ods:
         core_transport = ods['core_transport']['model']
         rho_TGLF = core_transport[0]['profiles_1d'][time_index]['grid_d']['rho_tor']
         # Qe
-        axes[0,1].plot(rho_TGLF, core_transport[2]['profiles_1d'][time_index]['electrons']['energy']['flux'], ls='-', marker="o", markersize=6, lw=2, label="total" + state_label)
-        axes[0,1].plot(rho_TGLF, core_transport[3]['profiles_1d'][time_index]['electrons']['energy']['flux'], ls='-', marker="x", markersize=8, lw=2, label="target" + state_label) # W/m^2
-        axes[0,1].plot(rho_core_prof, core_transport[4]['profiles_1d'][time_index]['electrons']['energy']['flux'], ls='--', lw=3, label="power_balance")
-        axes[0,1].set_ylabel('$Q_e$ (W/$m^2$)', fontsize='small')
+        axes[0,1].plot(rho_TGLF, core_transport[2]['profiles_1d'][time_index]['electrons']['energy']['flux'], ls='-', lw=2.5, color=color_label_dict.setdefault(plotting_label, 'r'), label="total" + plotting_label)
```
Comment (orso82) [inline omas/omas_plot.py:1166]: If it has to be generic it does not make too much sense to call it `rho_TGLF`

File: omas/omas_plot.py (hunk 1 above)
Comment (TimSlendebroek) [reply omas/omas_plot.py:1092]: I see, so I am guessing the difference is that I want to call the same function but now with a different time_index i.e:

``` python
    return ods_time_plot(profiles_fluxes, ods=ods, time_index=time_index, fig=fig, 

axes=axes,plotting_label=plotting_label, show_total_density=show_total_density, plot_zeff=plot_zeff)
``` 
Is there a nicer way to do this?

File: omas/omas_plot.py (hunk 2 above)
Comment (TimSlendebroek) [reply omas/omas_plot.py:1166]: makes sense changed it to rho_transport_model

**Comment by TimSlendebroek** [comment]
also can we add the {ods}.plot_profiles_fluxes() method to:
image

I tested and calling the .plot_profiles_fluxes() method on an ods works!
(Also it would be nice for that plotting method to grey out when core_transport is not avail)

**Comment by TimSlendebroek** [comment]
The thing is that in this way you can add a core_sample in any time_index spot as is done with ods_core_profiles for example. I added the core_profiles and equliubrium in the second commit. I am not sure why it is failing @orso82

**Comment by orso82** [comment]
@TimSlendebroek if you click on the `Details` on the right of the Travis CI and follow the white rabbit it will tell you why the regression tests are failing.

**Comment by TimSlendebroek** [comment]
I can see the error message now but it is odd since:
``` python
ods_test.sample_core_transport()
ods_test.sample_core_profiles()
ods_test.sample_equilibrium()
````
Should add all the samples I need (this is what I see when I call these methods in an omfit session)

**Comment by orso82** [comment]
@TimSlendebroek these are the errors that need to get addressed:
```
======================================================================
ERROR: test_satisfy_imas_requirements (test_omas_core.TestOmasCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/gafusion/omas/omas/tests/test_omas_core.py", line 296, in test_satisfy_imas_requirements
    ods.satisfy_imas_requirements()
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 1478, in satisfy_imas_requirements
    out.append(self.getraw(ds).satisfy_imas_requirements(attempt_fix=attempt_fix, raise_errors=raise_errors))
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 1504, in satisfy_imas_requirements
    raise ValueError(self.location + '.time cannot be automatically filled! Missing time information in the data structure.')
ValueError: core_sources.time cannot be automatically filled! Missing time information in the data structure.
======================================================================
ERROR: test_core_pressure (test_omas_plot.TestOmasPlot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/gafusion/omas/omas/tests/test_omas_plot.py", line 248, in test_core_pressure
    ods3.sample_core_profiles(add_junk_ion=True)
  File "/home/travis/build/gafusion/omas/omas/omas_sample.py", line 178, in core_profiles
    ions[len(ions) - 1][items] *= 0
NameError: name 'items' is not defined
======================================================================
ERROR: test_core_transport (test_omas_plot.TestOmasPlot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/gafusion/omas/omas/tests/test_omas_plot.py", line 238, in test_core_transport
    ods_test.plot_core_transport_fluxes()
  File "/home/travis/build/gafusion/omas/omas/omas_plot.py", line 1106, in core_transport_fluxes
    if ods['core_transport.ids_properties.comment'] == "TGYRO":
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 931, in __getitem__
    return value.__getitem__(key[1:], cocos_and_coords)
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 931, in __getitem__
    return value.__getitem__(key[1:], cocos_and_coords)
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 921, in __getitem__
    dynamic=self.dynamic))
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 643, in __setitem__
    raise ValueError('`%s` has no data' % location)
ValueError: `core_transport.ids_properties.comment` has no data
======================================================================
ERROR: test_satisfy_imas_requirements (test_omas_core.TestOmasCore)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/gafusion/omas/omas/tests/test_omas_core.py", line 296, in test_satisfy_imas_requirements
    ods.satisfy_imas_requirements()
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 1478, in satisfy_imas_requirements
    out.append(self.getraw(ds).satisfy_imas_requirements(attempt_fix=attempt_fix, raise_errors=raise_errors))
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 1504, in satisfy_imas_requirements
    raise ValueError(self.location + '.time cannot be automatically filled! Missing time information in the data structure.')
ValueError: core_sources.time cannot be automatically filled! Missing time information in the data structure.
======================================================================
ERROR: test_core_pressure (test_omas_plot.TestOmasPlot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/gafusion/omas/omas/tests/test_omas_plot.py", line 248, in test_core_pressure
    ods3.sample_core_profiles(add_junk_ion=True)
  File "/home/travis/build/gafusion/omas/omas/omas_sample.py", line 178, in core_profiles
    ions[len(ions) - 1][items] *= 0
NameError: name 'items' is not defined
======================================================================
ERROR: test_core_transport (test_omas_plot.TestOmasPlot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/gafusion/omas/omas/tests/test_omas_plot.py", line 238, in test_core_transport
    ods_test.plot_core_transport_fluxes()
  File "/home/travis/build/gafusion/omas/omas/omas_plot.py", line 1106, in core_transport_fluxes
    if ods['core_transport.ids_properties.comment'] == "TGYRO":
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 931, in __getitem__
    return value.__getitem__(key[1:], cocos_and_coords)
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 931, in __getitem__
    return value.__getitem__(key[1:], cocos_and_coords)
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 921, in __getitem__
    dynamic=self.dynamic))
  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 643, in __setitem__
    raise ValueError('`%s` has no data' % location)
ValueError: `core_transport.ids_properties.comment` has no data
----------------------------------------------------------------------
```

**Comment by TimSlendebroek** [comment]
hmm still some issues.

Still some issues with downsampling equilib.2d and sample_core_sources.

Also ['equilibrium']['time_slice'][0]['profiles_1d']['centroid'] appears in the OMFITtree but omas is telling me this doesn't exist in the IMAS datastructure (which is right checking the dataschema). Is there a place for these quantities:

image

in the IMAS data structure?

**Comment by orso82** [comment]
That's correct. We should place a request for adding the flux surfaces `centroid` to IMAS. It's not there right now. This is not what is holding up the regression tests though.

**Comment by TimSlendebroek** [comment]
Well I fixed the comment issue so now:
``` python
Traceback (most recent call last):

  File "/home/travis/build/gafusion/omas/omas/tests/test_omas_core.py", line 133, in test_coordinates

    ods = ods_sample()

  File "/home/travis/build/gafusion/omas/omas/omas_sample.py", line 40, in ods_sample

    ods = eval(ds)(ods, time_index=k)

  File "/home/travis/build/gafusion/omas/omas/omas_sample.py", line 138, in equilibrium

    ods['equilibrium.time_slice'][time_index].update(eq['equilibrium.time_slice.0'])

  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 1252, in update

    self[item] = ods2[item]

  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 801, in __setitem__

    self.getraw(key[0])[key[1:]] = pass_on_value

  File "/home/travis/build/gafusion/omas/omas/omas_core.py", line 670, in __setitem__

    raise LookupError(underline_last(text, len('LookupError: ')) + '\n' + options)

LookupError: Not a valid IMAS 3.28.1 location: equilibrium.time_slice.0.profiles_1d.centroid

                                                                                    ^^^^^^^^
```

**Comment by TimSlendebroek** [comment]
I think it is ready now apart from the centroid problem.

**Comment by TimSlendebroek** [comment]
I guess we need to add a wall ODS structure to sample_equilib

**Comment by orso82** [comment]
@TimSlendebroek please note that I have added a `make samples` which is what I want to use to generate the sample json files. Give me a second to fix what I have done there...

**Comment by TimSlendebroek** [comment]
:)

**Comment by orso82** [comment]
yes, progress. Note: One thing that we need to do throughout OMFIT is to start using `rotation_frequency_tor_sonic` instead of `omega0`

**Pull request merged**